### PR TITLE
React js

### PR DESCRIPTION
**Pull request by maurerle**
I barely have experience with react JS and the grafana sdk, but I somehow managed to make this work with the help of @intermittentnrg from #19 

This is just a draft to provide some substance of fixing things :)

I don't know how one can work around loading the initial dashboard without losing the javascript context. maybe someone can help here.

Note that there is also `this.grafanaRuntime.getDashboardTimeRange()` and stuff like this.

### CODE REVIEW

**Comment by intermittentnrg** [comment]
Great work!

Would prefer it if it was split into multiple smaller PRs tho. I'll leave some comments.

Btw I saw you work with energy data, have you checked out my grafana website? I have data for ~170 areas and all code is open source on github.

File: grafanimate/grafana-studio.js [hunk 1]
```diff
@@ -126,14 +114,14 @@ class GrafanaStudioSrv {
 
   waitForDashboard(uid, resolve, reject) {
     log("waitForDashboard");
-    var dashboard = this.dashboardSrv.getCurrent();
+    var dashboard = __grafanaSceneContext._state;
     if (dashboard) {
       // Sanity check. Has the right dashboard been loaded actually?
       if (dashboard.uid == uid) {
```
Comment (intermittentnrg) [inline grafanimate/grafana-studio.js:120]: I think this hack can be removed.

File: grafanimate/grafana-studio.js [hunk 2]
```diff
@@ -126,14 +114,14 @@ class GrafanaStudioSrv {
...
         // Quick hack to remove specific panel from specific dashboard.
         // FIXME
         if (uid == "DLOlE_Rmz") {
-          dashboard.panels.shift();
+          __grafanaSceneContext.getDashboardPanels().shift();
         }
```
Comment (intermittentnrg) [inline grafanimate/grafana-studio.js:125]: I think this hack can be removed.

**Comment by maurerle** [comment]
Hi @intermittentnrg this is not really meant as a PR which should get merged, as there are still various things to be fixed:

* login does not work
* changing the view of the dashboard dynamically does not work
* hooking into events was not tested
* I don't really know which parts of the code should be removed now
* Without this 7f44ea355f1a95bab6686f348183ef30cc008d01 - the resulting output has a far too high height, which should be fixed

**Comment by intermittentnrg** [comment]
Awesome. Yes Timescale sponsored me with free Timescale cloud with I share with Frank Boerman.
The idea is that while our projects are completely separate we can at least query each others tables.

But I used US location cause it's slightly cheaper and I optimize cloud bills even when someone else pays the bill, and Frank Boerman insisted on EU location. I will migrate to same server eventually but it's likely at least a days work to copy all data and sync etc, so for now we only share cloud account, not instance.

Yes I need to update Grafana. Working on other things also. I've been posting price map videos by driving browser and filling in timepicker and posting to https://x.com/IntermittentNRG

Feel free to DM on X, I also have a energy grafana club group chat on X which it sounds like you should be a part of, tho it's mostly inactive. I spotted a bunch of instances using grafana for energy.


As for the PR I think some changes you made are great and should be merged. If you split them up I'm happy to merge, @amotl gave me permission to merge for this repo. Or I could split them myself?

**Comment by maurerle** [comment]
Oh cool, here it is the university institute hosting it on a VM with 64GB RAM and 48 vCPU for everything, running on timescaleDB as well :blush: 

Very cool! :)
I don't have an X Account though, but thank you!

If you have a little time, it would be great to check into the open issues from https://github.com/grafana-toolbox/grafanimate/pull/30#issuecomment-2499982962

Feel free to split this up and adapt it, unfortunately I do not have that much knowledge of reactJS so I could need some help with this.
I mean, this project currently works well for grafana versions <11

After merging, it would only work with issues for Grafana >=11.3 - so we should at least get good support for latest Grafana :|

**Review by amotl — APPROVED** [review]
Thank you so much for this excellent refurbishment patch. 💯

> After merging, it would only work with issues for Grafana >=11.3 - so we should at least get good support for latest Grafana :|.

I think we should merge without much ado, there is no need to be too anal about backward-compatibility or otherwise holding back any progress on making this software work ~~well~~ reasonably again.

All flaws and caveats can be documented and tackled on behalf of subsequent patches, and I am looking forward to see them coming. 🍀

**Comment by maurerle** [comment]
I am fine with merging this now.

**Comment by intermittentnrg** [comment]
Changes look good I have no nitpicks / comments to add! Of course there's more changes to make in further PRs.

Also can  dashboard uid can be removed and just use the URL?
I never understood why I have to provide the uid in both the URL and as separate parameter.

**Comment by maurerle** [comment]
The dashboard uid has to be provided individually as there are two modes:
Dahsboard mode (with /d/ inbetween)
Panel mode (with /d-solo/ inbetween)

So thats why it is needed that way :)

**Comment by intermittentnrg** [comment]
Hmm OK, I just make custom copies of my dashboards for use with grafanimate so I never even noticed that there's a panel mode.

Good job!

**Pull request merged**